batchflow/utils_import.py

```diff
@@ -0,0 +1,128 @@
+""" Utility functions for delayed / missing imports. """
+import os
+import sys
+from importlib import import_module
+
+
+
+class DelayedImport:
+    """ Proxy class to postpone import until the first access.
+    Useful, when the import takes a lot of time (PyTorch, TensorFlow, ahead-of-time compiled functions).
+
+    Note that even access for module inspection (like its documentation or dir) would trigger module loading.
+
+    Examples::
+        >>> from .xxx import yyy                                                              # before
+        >>> yyy = DelayedImport(module='.xxx', package=__name__, attribute='yyy')             # after
+
+    Parameters
+    ----------
+    module : str
+        Name of module to load.
+    package : str, optional
+        Anchor for resolving the package name. Used only for relative imports.
+    attribute : str, optional
+        Name of attribute to get from loaded module.
+    help : str, optional
+        Additional help on import errors.
+    """
+    __file__ = globals()["__file__"]
+    __path__ = [os.path.dirname(__file__)]
+
+    def __init__(self, module, package=None, attribute=None, help=None):
+        self.module, self.package, self.attribute = module, package, attribute
+        self.help = help
+        self._loaded_module = None
+
+    @property
+    def loaded_module(self):
+        """ Try loading the module at the first access. """
+        if self._loaded_module is None:
+            try:
+                self._loaded_module = import_module(self.module, self.package)
+                if self.attribute is not None:
+                    self._loaded_module = getattr(self._loaded_module, self.attribute)
+            except ImportError as e:
+                if self.help:
+                    raise ImportError(f"No module named '{self.module.strip()}'! {self.help}") from e
```

Comment:
Do we really need `strip`? As it might conceal some typos in the module name which would be hard to detect.